src/solver/conv_winoRxS_ultra_f2x3.cpp

```diff
@@ -0,0 +1,706 @@
+        && problem.in_layout == "NCHW"))
+        return false;
+    // clang-format on
+
+    const auto n_inputs_per_group  = problem.n_inputs / problem.group_counts,
+               n_outputs_per_group = problem.n_outputs / problem.group_counts;
+
+    return IsShaderContraintsMet(problem.kernel_size_h, // RxS
+                                 problem.kernel_size_w,
+                                 problem.kernel_stride_h,
+                                 problem.kernel_stride_w,
+                                 n_inputs_per_group,  // C
+                                 n_outputs_per_group, // K
+                                 problem.in_height,   // HxW
+                                 problem.in_width,
+                                 problem.out_height, // OHxOW
+                                 problem.out_width,
+                                 problem.batch_sz, // N
+                                 ctx,
+                                 problem);
+#else
+    std::ignore = ctx;
+    return false;
+#endif
+}
+
+size_t ConvBinWinogradUltraRxSf2x3::GetWorkspaceSize(const ProblemDescription& problem) const
+{
+    const auto desc          = WinogradUltraDescription(problem);
+    const size_t num_tiles_H = Ceil(desc.out_H, o_tile_H);
+    const size_t num_tiles_W = Ceil(desc.out_W, o_tile_W);
+    const size_t N           = desc.N;
+    const size_t works       = desc.flush == flush_control::FLUSH_N
+                                   ? N * Ceil(num_tiles_H * num_tiles_W, group_size)
+                                   : Ceil(N * num_tiles_H * num_tiles_W, group_size);
+
+    return works * group_size * sizeof(uint32_t);
+}
+
+ConvSolution ConvBinWinogradUltraRxSf2x3::GetSolution(const ExecutionContext& ctx,
+                                                      const ProblemDescription& problem) const
+{
+    const unsigned n_groups = ctx.GetStream().GetMaxHardwareComputeUnits();
+    const auto desc         = WinogradUltraDescription(problem);
+    const auto intl_factor  = 1;
+
+    uint64_t reserved_offset = 0;
+    int* reserved_ptr        = nullptr;
+    float relu_alpha         = 1.0;
+
+    std::vector<uint32_t> control_buf;
+    WU_control_make_3x3(desc, control_buf, n_groups, intl_factor);
+
+    const unsigned n_works     = control_buf.size() / group_size;
+    const size_t workspace_req = GetWorkspaceSize(problem);
+    const size_t workspace_buf = control_buf.size() * sizeof(decltype(control_buf)::value_type);
+
+    if(workspace_req != workspace_buf)
+        MIOPEN_THROW("ConvBinWinogradUltraRxSf2x3 invalid control buffer size");
+
+    const size_t wg_size = 256;
+
+    KernelInfo kernel;
+
+    kernel.g_wk.push_back(wg_size * n_groups * problem.group_counts);
+    kernel.g_wk.push_back(1);
+    kernel.g_wk.push_back(1);
+
+    kernel.l_wk.push_back(wg_size);
+    kernel.l_wk.push_back(1);
+    kernel.l_wk.push_back(1);
+
+    std::string kernel_name    = "miopenSp3AsmConv_Ultra_v1_0_14_gfx10";

```

Comment:
These should probably be `constexpr` or at least `const`.

Reply:
Thank you! Fixed in [306bbff](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1510/commits/306bbff8caa62c1f52ecbdbe6a65e435362f6ae1).